redaxo/src/addons/media_manager/lib/effects/effect_convert2img.php

```diff
@@ -72,9 +72,19 @@ public function execute()
 
         $to_path = rex_path::addonCache('media_manager', 'media_manager__convert2img_' . md5($this->media->getMediaPath()) . '_' . $filename_wo_ext . $convert_to['ext']);
 
-        $cmd = $convert_path . ' -density '.$density.' "' . $from_path . '[0]" -colorspace RGB "' . $to_path . '"';
-
-        exec($cmd, $out, $ret);
+		if ($convert_path == 'imagick') {
```

Comment:
hier fehlt ein "if-extension-loaded"

Reply:
Warum?
$convert_path liefert in meiner Änderung "imagick" zurück.

Reply:
eine methode die `getConvertPath()` pass heißt sollte nichts liefern, was nichts mit `convert` zu tun hat.

vor deiner Änderung hat `getConvertPath` einen file-system pfad zurückgegeben wo das `convert` commando zu finden ist.

nach deinen Änderung gibt es jetzt noch einen "magic string" `imagick` der nix mit der urspr. intention der methode zu tun hat (nämlich einen filesystem pfad zu returnen) sondern nur noch als "Kennzeichen" missbraucht wird

Reply:
Der Wert ("Kennzeichen") kommt aber nur, wenn imagick vorhanden wäre, sonst der übliche Pfad.

Reply:
ich glaube, @staabm will darauf hinaus, dass man sowas besser nicht macht.